### PR TITLE
Base.convert method to translate a `NonlinearFunction` to an `ODEfunction`

### PR DESCRIPTION
**Pull request by frankschae**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This should get a unit test.

File: src/scimlfunctions.jl [hunk 1]
```diff
@@ -3890,3 +3890,28 @@ for S in [:ODEFunction
...
+"""
+function Base.convert(::Type{ODEFunction}, f::NonlinearFunction)
+    iip = isinplace(f)
+    _f = iip ? (du, u, p, t) -> (f.f(du, u, p); nothing) : (u, p, t) -> f.f(u, p)
+
+    ODEFunction{iip}(_f;
+                     mass_matrix = f.mass_matrix,
+                     analytic = f.analytic,
+                     tgrad = f.tgrad,
+                     jac = f.jac,
```
Comment (ChrisRackauckas) [inline src/scimlfunctions.jl:3907]: this would need to add a `t` to it as well?

File: src/scimlfunctions.jl [hunk 2]
```diff
@@ -3890,3 +3890,28 @@ for S in [:ODEFunction
...
+Converts a NonlinearFunction into a ODEFunction.
+"""
+function Base.convert(::Type{ODEFunction}, f::NonlinearFunction)
+    iip = isinplace(f)
+    _f = iip ? (du, u, p, t) -> (f.f(du, u, p); nothing) : (u, p, t) -> f.f(u, p)
+
+    ODEFunction{iip}(_f;
+                     mass_matrix = f.mass_matrix,
+                     analytic = f.analytic,
+                     tgrad = f.tgrad,
```
Comment (ChrisRackauckas) [inline src/scimlfunctions.jl:3906]: add `t`?

File: src/scimlfunctions.jl [hunk 3]
```diff
@@ -3890,3 +3890,28 @@ for S in [:ODEFunction
...
+                     jac = f.jac,
+                     jvp = f.jvp,
```
Comment (ChrisRackauckas) [inline src/scimlfunctions.jl:3908]: t

File: src/scimlfunctions.jl [hunk 4]
```diff
@@ -3890,3 +3890,28 @@ for S in [:ODEFunction
...
+                     vjp = f.vjp,
```
Comment (ChrisRackauckas) [inline src/scimlfunctions.jl:3909]: t

File: src/scimlfunctions.jl (hunk 1 above)
Comment (frankschae) [reply src/scimlfunctions.jl:3907]: oh yeah, good catch. I'll add it with some tests.

**Pull request merged**